rotkehlchen/chain/ethereum/modules/convex/convex_cache.py

```diff
@@ -0,0 +1,135 @@
+from rotkehlchen.db.drivers.gevent import DBCursor
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.globaldb.cache import (
+    globaldb_get_general_cache_values,
+    globaldb_get_unique_cache_value,
+    globaldb_set_general_cache_values,
+    globaldb_set_unique_cache_value,
+)
+from rotkehlchen.globaldb.handler import GlobalDBHandler
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import CacheType, ChecksumEvmAddress
+from rotkehlchen.utils.misc import hex_or_bytes_to_address
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.ethereum.node_inquirer import EthereumInquirer
+    from rotkehlchen.db.dbhandler import DBHandler
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class ConvexPoolData(NamedTuple):
+    pool_address: ChecksumEvmAddress
+    pool_name: str
+
+
+def get_existing_pools(cursor: DBCursor) -> set[ChecksumEvmAddress]:
+    return {
+        string_to_evm_address(address) for address in globaldb_get_general_cache_values(
+            cursor=cursor,
+            key_parts=(CacheType.CONVEX_POOL_ADDRESS,),
+        )
+    }
+
+
+def save_convex_data_to_cache(
+        write_cursor: DBCursor,
+        database: 'DBHandler',  # pylint: disable=unused-argument
+        new_data: list[ConvexPoolData],
+) -> None:
+    """Stores data about convex pools and their names in the global db cache tables."""
+    for pool in new_data:
+        globaldb_set_general_cache_values(
+            write_cursor=write_cursor,
+            key_parts=(CacheType.CONVEX_POOL_ADDRESS,),
+            values=[pool.pool_address],
+        )
+        globaldb_set_unique_cache_value(
+            write_cursor=write_cursor,
+            key_parts=(CacheType.CONVEX_POOL_NAME, pool.pool_address),
+            value=pool.pool_name,
+        )
+
+
+def read_convex_data_from_cache() -> dict[ChecksumEvmAddress, str]:
+    """Reads convex pools and names from global db cache tables."""
+    pools: dict[ChecksumEvmAddress, str] = {}
+    with GlobalDBHandler().conn.read_ctx() as cursor:
+        for pool_address in get_existing_pools(cursor):
+            if (pool_name := globaldb_get_unique_cache_value(
+                cursor=cursor,
+                key_parts=(CacheType.CONVEX_POOL_NAME, pool_address),
+            )) is None:
+                continue
+            pools[pool_address] = pool_name
+
+    return pools
+
+
+def query_convex_data_from_chain(ethereum: 'EthereumInquirer') -> dict[ChecksumEvmAddress, str]:
+    """
+    Query Booster contract and fetch all reward pools.
+    May raise:
+    - RemoteError if failed to query chain
+    """
+    booster_contract = ethereum.contracts.contract(BOOSTER)
+    pools_count = booster_contract.call(
+        node_inquirer=ethereum,
+        method_name='poolLength',
+    )
+    calls_to_booster = [(
+        booster_contract.address,
+        booster_contract.encode('poolInfo', [x]),
+    ) for x in range(pools_count)]
```

Comment:
maybe this code can be improved? If I undertood correctly this query all the pools all the time but we might have already some queried and in the globaldb right? Maybe you could store something like `LAST_QUERIED_CONVEX_POOL_INDEX` and then just query after those. I'm not sure how the logic goes from here... if everything gets deleted or what but it seems that we are querying everything each time